Feature Flags: Removed unused flags

psealock opened the pull request:
In cleaning up the disabling of flags, I noted some flags that weren't being used, so this PR cleans that up.

Its important to note that flags with `true` across all configs can be used to initialise features or be selectively turned off by outside extensions (such as wp-calypso-bridge) so its important to leave them.

### Question

Flags `wcpay` and `wcpay/support-international-countries` are turned on in every environment and the features seem mature enough that we likely won't revert them. Is there any objection to removing them as well?

### Detailed test instructions:

1. This PR only deals with the Analytics dashboard, so poke around WooCommerce > Analytics > Overview to ensure things still work.

**NOTE** If you disable Analytics, the Homescreen will not work because the performanceIndicators use endpoints that are disabled. This is a separate issue addressed in https://github.com/woocommerce/woocommerce-admin/pull/7234

psealock committed:
remove analytics-dashboard/customizable

psealock committed:
remove analytics-dashboard

psealock committed:
changelog

joshuatf commented on client/dashboard/dashboard-charts/index.js:
```diff
@@ -163,20 +163,16 @@ const DashboardCharts = ( props ) => {
 						onToggleHiddenBlock,
 					} ) }
-					{ window.wcAdminFeatures[
-						'analytics-dashboard/customizable'
```
I think I know the answer to this, but just to double check- would we now consider this stable enough to be part of the `analytics` feature flag?

joshuatf commented on config/core.json:
```diff
@@ -2,8 +2,6 @@
 	"features": {
 		"activity-panels": true,
 		"analytics": true,
-		"analytics-dashboard": true,
```
Same question here- would the `analytics-dashboard` be included in `analytics` or could these ever be considered separate toggleable features?

joshuatf reviewed (commented):
Thanks for tackling this clean up, Paul!

>Flags wcpay and wcpay/support-international-countries are turned on in every environment and the features seem mature enough that we likely won't revert them. Is there any objection to removing them as well?

While we may want to remove and/or update those feature flags, the reason listed ( " mature enough that we likely won't revert them" ) I think aligns more with the previous usage of feature flags and not the plan going forward.

We have some discussion on this somewhere, but maybe we need another p2 since there seems to be a lot of confusion around this.  The current plan is to use feature flags initially for gating, but also for encapsulation of a feature to turn it on or off.  We're hoping to encourage turning off specific features of WooCommerce Admin instead of using the filter that turns off everything.

The changes in this PR may still align with this goal, I just wanted to toss this out there for thought.

psealock commented:
Thanks for taking a look @joshuatf. 

Here is the discussion and proposed plan: pcShBQ-1i#comment-181 

So far a toggle was introduced [to turn off Analytics](https://github.com/woocommerce/woocommerce-admin/pull/7168) using the same mechanism as Navigation. This also [turns off associated API's and data stores](https://github.com/woocommerce/woocommerce-admin/pull/7193). 

We've been using flags as a development tool to avoid shipping features that are in progress, and I don't think we should stop doing that but it doesn't mean all features will be exposed as being able to be turned off, ie Homescreen. Ideally, that flag will be removed as well. At the moment its required because its PHP class isn't loaded without the flag being defined in the config.

Specifically addressing the comments here about these flags being folded into `analytics` - Yes, these are essentially already behind the analytics flag as a result of the two pull requests linked above.

>Same question here- would the analytics-dashboard be included in analytics or could these ever be considered separate toggleable features?

I don't think the desire is to offer a granular ability to toggle items and its already disabled if `analytics` is turned off.

joshuatf reviewed (approved):
>I don't think the desire is to offer a granular ability to toggle items and its already disabled if analytics is turned off.

Agreed on this; just wanted to sanity check the intent for the feature.  Thanks for the clean up around these feature flags- LGTM!  🚢

psealock committed:
Merge branch 'main' into fix/unused-feature-flags
